File: backend/backend/src/main/java/com/jj/backend/entity/Technology.java
package com.jj.backend.entity;

import com.jj.backend.config.TechnologyType;
import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "technologies")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Technology {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @Column(name = "name", nullable = false, unique = true)
    private String name;

    @Column(name = "type", nullable = false)
    private TechnologyType type;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

}

File: frontend/src/models/technology.ts
export type TechnologyType = "FRONTEND" | "BACKEND" | "DEVOPS" | "DATABASE";

export type TechnologyRequest = {
  name: string;
  type: TechnologyType;
};

export type TechnologyResponse = {
  id: number;
  name: string;
  type: TechnologyType;
  createdAt: string;
  updatedAt: string;
};

export type Technology = {
  id: number;
  name: string;
  type: TechnologyType;
};
